### DISCUSSION TITLE
[zksolc]<Error in plugin @matterlabs/hardhat-zksync-solc: Could not download zksolc compiler version info file>

### DISCUSSION THREAD

**Discussion by Madhav-Gupta-28** [zkSync-Community-Hub/zksync-developers#149] [Compilers] [answered]
### Which compiler are you using?

- [X] zksolc
- [ ] zkvyper

### Which version of Solidity or Vyper?

0.8.8

### zksolc/zkvyper version?

^0.4.1

### Explain the issue

```bash
I have Forked This Repo : https://github.com/billyjitsu/zk-paymaster-dapi-vip

And Then Try to Run in my Local Computer. 
I have install Dependencies by "yarn install" , but when compiling contracts by "yarn compile". 
I am getting this error :

yarn compile
yarn run v1.22.19
$ hardhat compile
Error in plugin @matterlabs/hardhat-zksync-solc: Could not download zksolc compiler version info file. Please check your internet connection and try again.

For more info run Hardhat with --show-stack-traces
error Command failed with exit code 1.
```

### Link to your repo?

https://github.com/billyjitsu/zk-paymaster-dapi-vip/

**Comment by dutterbutter** [comment]
Hmm might be the version, can you please update to latest. Can you try uninstalling:
```
yarn uninstall @matterlabs/hardhat-zksync-solc
``` 
Installing:
```
yarn install @matterlabs/hardhat-zksync-solc
```

**Reply by Madhav-Gupta-28** [reply]
I have done that. But The Error Persists

**Reply by dutterbutter** [reply]
hmm.... I just cloned and installed the dependencies fine on the provided repo? What os are you using?

**Comment by Madhav-Gupta-28** [comment]
Windows 10 ser

**Reply by MexicanAce** [reply]
Hi @Madhav-Gupta-28 👋 

Starting from the repo you linked, I was able to successfully run `yarn compile` on my Windows 10 machine. Could you please delete your `node_modules` directory and try `yarn install` again?

Here's my output for `yarn compile`:

```bash
PS C:\git\zk-paymaster-dapi-vip> yarn compile
yarn run v1.22.19
$ hardhat compile
√ Help us improve Hardhat with anonymous crash reports & basic usage data? (Y/n) · n
Downloading zksolc 1.3.14
zksolc version 1.3.14 successfully downloaded
Compiling contracts for zkSync Era with zksolc v1.3.14 and solc v0.8.8
Downloading solc 0.8.8
Compiling 43 Solidity files
Warning: This declaration shadows an existing declaration.
  --> @matterlabs/zksync-contracts/l2/system-contracts/libraries/Utils.sol:85:9:
   |
85 |         uint256 bytecodeLenInWords = _bytecode.length / 32;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
Note: The shadowed declaration is here:
  --> @matterlabs/zksync-contracts/l2/system-contracts/libraries/Utils.sol:43:5:
   |
43 |     function bytecodeLenInWords(bytes32 _bytecodeHash) internal pure returns (uint256 codeLengthInWords) {
   |     ^ (Relevant source part starts here and spans across multiple lines).


Warning: Unused function parameter. Remove or comment out the variable name to silence this warning.
  --> contracts/MyPaymaster.sol:59:32:
   |
59 |         returns (bytes4 magic, bytes memory context)
   |                                ^^^^^^^^^^^^^^^^^^^^


Warning: Unused local variable.
  --> contracts/MyPaymaster.sol:74:45:
   |
74 |             (address token, uint256 amount, bytes memory data) = abi.decode(
   |                                             ^^^^^^^^^^^^^^^^^


Successfully compiled 43 Solidity files
Done in 29.04s.
```

**Comment by GainLee** [comment]
I got this on my MacOS too.
I meet it on https://era.zksync.io/docs/dev/tutorials/gated-nft-paymaster-tutorial.html#implementing-validateandpayforpaymastertransaction-function
Need some help pls

**Comment by idea404** [comment, marked as answer]
@Madhav-Gupta-28 can you confirm you are still receiving this error? 

The plugin was throwing this error for all of its invocations previously because it was fetching this info file from a GitHub repository which had rate limits in place, the most recent release of this plugin (`1.0.2`) addresses this issue.

**Comment by longbipubacr** [comment]
It seems that the error is related to the @matterlabs/hardhat-zksync-solc plugin being unable to download the zksolc compiler version info file. This could be due to a network issue or an incorrect configuration.

To resolve this issue, ensure that your internet connection is stable and that there are no firewall or network restrictions blocking the download. Additionally, verify that your Hardhat configuration is correctly set up to use the @matterlabs/hardhat-zksync-solc plugin and that the plugin version is compatible with your Hardhat version.

You can also try running the command again after ensuring a stable internet connection and checking your configuration settings. If the issue persists, consider reaching out to the plugin maintainers for further assistance or troubleshooting steps.

**Comment by heuresfanegarnold** [comment]
The error indicates that the zksolc compiler version info file could not be downloaded, possibly due to a network issue. Please ensure that you have a stable internet connection and try running the compilation command again. Additionally, make sure that your firewall settings or any other security measures are not blocking the download process. If the issue persists, you may want to check if there are any ongoing network disruptions or try using a VPN to troubleshoot the connectivity problem.

**Comment by yamyv** [comment]
Issue: Error downloading zksolc compiler version info file when compiling contracts using Hardhat.

Solution: Check your internet connection and try again. If the issue persists, ensure that the zksolc compiler version specified in the configuration is correct and accessible.

**Comment by shipefred** [comment]
Which compiler are you using: zksolc
Version of Solidity or Vyper: 0.8.8
zksolc version: ^0.4.1
Issue Explanation: Encounter error while compiling contracts due to the inability to download zksolc compiler version info file.
Repo Link: https://github.com/billyjitsu/zk-paymaster-dapi-vip/

**Comment by wadejackie** [comment]
The user is encountering an error when attempting to compile contracts in a forked repository using the command "yarn compile". The error message indicates that the plugin @matterlabs/hardhat-zksync-solc is unable to download the zksolc compiler version info file. The user has provided a link to their repository. Further investigation may be needed to determine the cause of the error and potential solutions.
